### PR TITLE
Makes odo work with v0.3.0 of Service Binding Operator

### LINKED ISSUE #3973: Make "odo link" compatible with the breaking changes introduced by Service Binding Operator
## Which functionality do you think we should add?
The Service Binding Operator project introduced breaking changes in https://github.com/redhat-developer/service-binding-operator/pull/608 which will cause the "odo link" between a component and an Operator backed service to fail. odo is not compatible with the breaking changes introduced in the Operator.

## Why is this needed?
It is needed to make sure users can upgrade to a version of odo that is compatible with this breaking change.

## Acceptance Criteria

- [ ] Check if the cluster supports resources of type "ServiceBinding" using the GVK/R info.
- [ ] Create "ServiceBinding" instead of "ServiceBindingRequest" if it supports.
- [ ] Maintain backward compatibility for alpha version of SBO by keeping support for "ServiceBindingRequest". 
- [ ] Integration Tests

/area service-operators
/area component
/kind feature

### PR DESCRIPTION
**Pull request by dharmit**
**What type of PR is this?**
/kind api-change
/kind documentation

**What does this PR do / why we need it**:
This PR makes odo compatible with the changes in Service Binding Operator API.

**Which issue(s) this PR fixes**:

Fixes #3973 

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [x] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
1. CI should just pass.
1. Refer the steps in the documentation changes made with this PR and make sure things work

### CODE REVIEW

**Comment by dharmit** [comment]
```
level=error
level=error msg="Error: Error creating VPC: VpcLimitExceeded: The maximum number of VPCs has been reached."
level=error msg="\tstatus code: 400, request id: 0a165f0f-5583-488d-850b-20637ffe5d3f"
level=error
level=error msg="  on ../tmp/openshift-install-387733094/vpc/vpc.tf line 6, in resource \"aws_vpc\" \"new_vpc\":"
level=error msg="   6: resource \"aws_vpc\" \"new_vpc\" {"
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
2020/12/03 06:14:38 Failed to upload $KUBECONFIG: timed out waiting for the condition: stat /tmp/secret/kubeconfig: no such file or directory
error: failed to execute wrapped command: exit status 1 
```
Link: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_odo/4274/pull-ci-openshift-odo-master-v4.6-integration-e2e/1334376747369500672

File: docs/public/install-service-binding-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+=== Installing Service Binding Operator on Kubernetes
+
+Steps mentioned in this section were tested on link:https://minikube.sigs.k8s.io/[minikube]. We tested this on minikube v1.15.1 but it should work on v1.11.0 and above.
+
+For Kubernetes, Service Binding Operator is not yet available via the OLM. The team is link:https://github.com/redhat-developer/service-binding-operator/issues/727[working on making it available].
+
+To install the Operator, execute the following `kubectl` command:
+[source,sh]
+----
+$ kubectl apply -f https://gist.githubusercontent.com/dharmit/0e05be20e98c9271b2117acea7908cc2/raw/1e45fc89fc576e184e41fcc23e88d35f0e08a7e9/install.yaml
```
Comment (adisky) [inline docs/public/install-service-binding-operator.adoc:22]: could we move this in odo repo?

Comment (dharmit) [reply docs/public/install-service-binding-operator.adoc:22]: We would actually get rid of it very soon once it's added into Service Binding Operator repo. https://kubernetes.slack.com/archives/C019LQYGC5C/p1606820080016700

**Comment by adisky** [comment]
Mostly looks good to me, couldn't test it as struggling with cluster setup.

**Pull request merged**